src/utils/arfs_builders/arfs_drive_builders.ts

```diff
@@ -70,11 +69,11 @@ export class ArFSPublicDriveBuilder extends ArFSMetadataEntityBuilder<ArFSPublic
 			this.appVersion?.length &&
 			this.arFS?.length &&
 			this.contentType?.length &&
-			this.driveId?.length &&
+			this.driveId &&
 			this.entityType?.length &&
 			this.txId?.length &&
 			this.unixTime &&
-			this.driveId == this.entityId &&
+			this.driveId.equals(this.entityId as EntityID) &&
```

Comment:
Why is this check in the public drive builder but not the private? Do we need this?

Reply:
befad39